beets/dbcore/db.py

```diff
@@ -668,8 +668,17 @@ def mutate(self, statement, subvals=()):
         """Execute an SQL statement with substitution values and return
         the row ID of the last affected row.
         """
-        cursor = self.db._connection().execute(statement, subvals)
-        return cursor.lastrowid
+        try:
+            cursor = self.db._connection().execute(statement, subvals)
+            return cursor.lastrowid
+        except sqlite3.OperationalError as exe:
+            if exe.message == "unable to open database file":
```

Comment:
Just to check: is this the best way to disambiguate the kind of OperationalError? I believe it might be, but it's worth double-checking whether the `sqlite3` module has error codes or anything.